PYME/Acquire/Hardware/pco/pco_sdk_cam.py

```diff
@@ -155,24 +161,74 @@ def _poll_loop(self):
                         # wait for the buffer
                         wait_status = k32_dll.WaitForSingleObject(self._buf_event[_curr_buf], self._timeout)
-                        if wait_status:
-                            logger.warning(f"Waited too long for buffer ({self._timeout} ms).") 
-                            
-                            #TODO: we currently continue as if we got the buffer - is this the right thing to do?
-                            # Presumably the status will be non-zero and we will drop the buffer?, but then what 
-                            # happens to those buffers? do they just dissapear?
+                        if wait_status == WAIT_TIMEOUT:
+                            # Timeout: do NOT deliver a frame; recycle the buffer
+                            self._n_timeouts += 1
+                            self._consecutive_errors += 1
+
+                            logger.warning(
+                                f"PCO: WaitForSingleObject timeout after {self._timeout} ms "
+                                f"(timeouts={self._n_timeouts}, consecutive_errors={self._consecutive_errors}) "
+                                f"on buffer {_curr_buf}. Recycling."
+                            )
+
+                            k32_dll.ResetEvent(self._buf_event[_curr_buf])
+                            self._buffers_to_queue.put(_curr_buf)
+
+                            # After a few consecutive timeouts, flag for higher-level recovery logic
+                            if self._n_timeouts >= 3:
+                                logger.error("PCO: repeated buffer timeouts -> hardware_overflowed=True")
+                                self.hardware_overflowed = True
                         
-                        k32_dll.ResetEvent(self._buf_event[_curr_buf])
-                        # make sure this buffer is safe to use
-                        status = self._buffer_status[_curr_buf]
-                        if status:
-                            logger.warning(f"Error {status} during check of buffer {_curr_buf}.")
-                            #DB: Do you see a lot of these warnings? IE - do we get one every time we have a timeout?
-                            # drop this buffer
+                        elif wait_status == WAIT_OBJECT_0:
+                            # Normal case
+                            self._n_timeouts = 0
+                            self._consecutive_errors = 0
+
+                            k32_dll.ResetEvent(self._buf_event[_curr_buf])
+
+                            status = self._buffer_status[_curr_buf]
+                            if status != 0:
+                                # Bad buffer -> recycle it, don't deliver
+                                logger.warning(f"PCO: error {status} during check of buffer {_curr_buf}. Recycling.")
+                                self._consecutive_errors += 1
+                                self._buffers_to_queue.put(_curr_buf)
```

Comment:
We might not want to re-queue bad buffers unless we are in continuous mode  (if we want to match the current logic, i.e.:

For non-continuous modes: 
- A single buffer is queued in TriggerAq
- We wait on this buffer, and return once full.

Re-queueing buffers that are bad risks letting the acquisition get out of sync with the triggers (the re-queued buffer might end up waiting on the next trigger)